lib/src/git.dart

```diff
@@ -121,13 +122,34 @@ String? repoRoot(String dir) {
   return null;
 }
 
+/// '--recourse-submodules' was introduced in Git 2.14
+/// (https://git-scm.com/book/en/v2/Git-Tools-Submodules).
+final _minSupportedGitVersion = Version(2, 14, 0);
+
 /// Checks whether [command] is the Git command for this computer.
 bool _tryGitCommand(String command) {
   // If "git --version" prints something familiar, git is working.
   try {
     var result = runProcessSync(command, ['--version']);
-    var regexp = RegExp('^git version');
-    return result.stdout.length == 1 && regexp.hasMatch(result.stdout.single);
+
+    if (result.stdout.length != 1) return false;
+    final output = result.stdout.single;
+    final match = RegExp(r'^git version (\d+)\.(\d+)\.').matchAsPrefix(output);
+
+    if (match == null) return false;
+    // Git seems to use many parts in the version number. We just check the
+    // first two.
+    final major = int.parse(match[1]!);
+    final minor = int.parse(match[2]!);
+    if (Version(major, minor, 0) < _minSupportedGitVersion) {
+      // We just warn here, as some features might work with older versions of
+      // git.
+      log.warning('''
+Pub needs at least git version 2.14 for all features to work.
+You seem to have ${output.substring('git '.length)}.
```

Comment:
We have a variable for topLevelCommand... 
```suggestion
You have a very old version of git (version ${output.substring('git '.length)}),
for $topLevelCommand it is recommanded to use git version 2.14 or newer.
```